src/ucs/debug/memtrack.c

```diff
@@ -285,6 +286,28 @@ static UCS_F_ALWAYS_INLINE void ucs_memtrack_releasing_internal(void *ptr)
     ucs_memtrack_do_releasing(ptr);
 }
 
+static void ucs_memtrack_vfs_read(void *obj, ucs_string_buffer_t *strb,
+                                  void *arg_ptr, uint64_t arg_u64)
+{
+    char *buffer;
+    size_t size;
+    FILE *f;
+
+    f = open_memstream(&buffer, &size);
+    ucs_memtrack_dump(f);
+    fclose(f);
```

Comment:
let's change ucs_memtrack_dump to accept string buffer to dump to

Reply:
If a user will pass string buffer based on a dynamic size array, then it will be a dead lock.
Thus, the interface method `ucs_memtrack_dump` should be with a strict restriction.
A restriction usually complicates the usage of an interface.

Reply:
ok